apps/drain-safe/src/App.tsx

```diff
@@ -0,0 +1,192 @@
+            value: item.balance,
+            data: '0x',
+          }
+        : {
+            // For other token types, generate a contract tx
+            to: web3Utils.toChecksumAddress(item.tokenInfo.address),
+            value: '0',
+            data: encodeTxData(erc20.transfer, toAddress, item.balance),
+          };
+    });
+
+    let safeTxHash = '';
+    try {
+      const data = await sdk.txs.send({ txs });
+      safeTxHash = data.safeTxHash;
+      console.log(safeTxHash);
+    } catch (e) {
+      console.error(e);
+    }
+
+    if (!safeTxHash) {
+      setSubmitting(false);
+      return;
+    }
+
+    const poll = setInterval(async (): Promise<void> => {
+      try {
+        const safeTx = await sdk.txs.getBySafeTxHash(safeTxHash);
+        console.log(safeTx);
+      } catch (e) {
+        setSubmitting(false);
+        return;
+      }
+
+      setSubmitting(false);
+      setFinished(true);
+
+      setAssets(
+        assets.map((item) => ({
+          ...item,
+          balance: '0',
+          fiatBalance: '0',
+        })),
+      );
+    }, 1000);
+
+    return () => {
+      clearInterval(poll);
+    };
+  }, [sdk, assets, toAddress]);
+
+  const onToAddressChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
+    setToAddress(e.target.value);
+  };
+
+  // Fetch balances
+  useEffect(() => {
+    fetchBalances();
+  }, [fetchBalances]);
+
+  return (
+    <Container>
+      <Title size="md">Drain Account</Title>
+
+      {isFinished && <Text size="lg">The transaction has been created. Refresh the app when it's executed.</Text>}
+
+      <Table
+        headers={[
+          { id: 'col1', label: 'Asset' },
+          { id: 'col2', label: 'Amount' },
+          { id: 'col3', label: `Value, ${CURRENCY}` },
+        ]}
+        rows={assets.map((item: Asset, index: number) => ({
+          id: `${index}`,
+          cells: [
+            {
+              content: (
+                <Flex>
+                  <Icon src={item.tokenInfo.logoUri} alt="" />
+                  {item.tokenInfo.name}
+                </Flex>
+              ),
+            },
+            { content: web3Utils.fromWei(item.balance) },
+            { content: item.fiatBalance },
+          ],
+        }))}
+      />
+
+      {submitting ? (
+        <>
+          <Flex centered>
+            <Loader size="md" />
+          </Flex>
+          <Flex centered>
+            <Button
+              size="lg"
+              color="secondary"
+              onClick={() => {
+                setSubmitting(false);
+              }}
+            >
+              Cancel
+            </Button>
+          </Flex>
+        </>
+      ) : (
+        <>
+          <TextField onChange={onToAddressChange} value={toAddress} label="Recipient" />
+
+          <Flex centered>
+            <Button
+              size="lg"
+              color="primary"
+              variant="contained"
+              onClick={submitTx}
```

Comment:
does it make sense to add a confirmation modal?

Reply:
no, it looks confusing because right after that safe-react will show a confirmation modal

Reply:
that's correct. I forgot about that.